src/main/java/org/broadinstitute/dsm/jobs/TestBostonUPSTrackingJob.java

```diff
@@ -36,10 +37,10 @@
     private static String SQL_AVOID_DELIVERD = " and (ups_tracking_status is null or ups_return_status is null or ups_tracking_status not like \"" + UPSStatus.DELIVERED_TYPE + " %\" or ups_return_status not like \"" + UPSStatus.DELIVERED_TYPE + " %\")" +
             " order by kit.dsm_kit_request_id ASC";
 
-    private static final String SQL_UPDATE_UPS_TRACKING_STATUS = "UPDATE ddp_kit SET ups_tracking_status = ?, ups_tracking_date = ? " +
-            "WHERE dsm_kit_id <> 0 and  tracking_to_id = ? and test_result is null";
-        private static final String SQL_UPDATE_UPS_RETURN_STATUS = "UPDATE ddp_kit SET ups_return_status = ?, ups_return_date = ? " +
-                "WHERE dsm_kit_id <> 0 and tracking_return_id= ? and test_result is null";
+    private static final String SQL_UPDATE_UPS_TRACKING_STATUS = "UPDATE ddp_kit kit, ddp_kit_request request SET ups_tracking_status = ?, ups_tracking_date = ? " +
```

Comment:
I think this should take the form `UPDATE ddp_kit_request where...` and avoid putting `ddp_kit` into the update statement.  I'm not sure mysql is smart enough to **not** put an update lock on `ddp_kit`.  It might be, but better to make the update as narrow as possible.

Also, are `tracking_to_id` and `tracking_return_id` indexed?  If not, please index them so that these update locks are held as briefly as possible.